### PR TITLE
feat(Dialog): a new optional prop for the close button

### PR DESCRIPTION
**Pull request by ITZSHOAIB**
### Overview

Added a new optional prop for the close button related to its tip.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [ENT-233](https://skillsoftdev.atlassian.net/browse/ENT-233)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

- Run `yarn start`
- Visit http://localhost:6006/?path=/docs/molecules-modals-dialog--dialog
- A new prop `closeButtonTipText` should be visible in the Code Playground section

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| LE Enterprise       | [LE Enterprise Link](https://github.com/codecademy-engineering/mono/pull/6987)  | [LE Enterprise Env](https://le-ent-pr-6987.dev-eks.codecademy.com/emodules/emod-javascript-requests-with-fetch-api)   |

[ENT-233]: https://skillsoftdev.atlassian.net/browse/ENT-233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/gamut/src/Modals/Dialog.tsx [hunk 1]
```diff
@@ -69,12 +73,12 @@ export const Dialog: React.FC<DialogProps> = ({
         </Text>
         <Box gridArea="close">
           <IconButton
-            aria-label="Close Dialog"
+            aria-label={closeButtonLabel}
```
Comment (dreamwasp) [inline packages/gamut/src/Modals/Dialog.tsx:76]: i think you can delete this line, i don't think it needs an aria-label, the tip text should be enough (it auto-adds aria-label based on tip text)

Comment (ITZSHOAIB) [reply packages/gamut/src/Modals/Dialog.tsx]: Updated in the latest commit. Thanks!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.0-alpha.b67490.0<br/>@codecademy/gamut-kit@0.6.430-alpha.b67490.0<br/>@codecademy/styleguide@67.0.5-alpha.b67490.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66aa6950474e0f1136b8beed--gamut-preview.netlify.app>https://66aa6950474e0f1136b8beed--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66aa6950474e0f1136b8beed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.0-alpha.2b7787.0<br/>@codecademy/gamut-kit@0.6.430-alpha.2b7787.0<br/>@codecademy/styleguide@67.0.5-alpha.2b7787.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66ab2e805dabdd880ef0e99b--gamut-preview.netlify.app>https://66ab2e805dabdd880ef0e99b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66ab2e805dabdd880ef0e99b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.0-alpha.88b31f.0<br/>@codecademy/gamut-kit@0.6.431-alpha.88b31f.0<br/>@codecademy/styleguide@67.0.6-alpha.88b31f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66ab88fbb76dd62564f3891d--gamut-preview.netlify.app>https://66ab88fbb76dd62564f3891d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66ab88fbb76dd62564f3891d>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
this looks great! could you open a PR with the `gamut-kit` alpha package? there are instructions [here](https://www.notion.so/codecademy/Bumping-gamut-kit-in-the-monorepo-b6107ee2b38f44cca176a967bdf67747#60ec17293542435cb56fb19f7679eea9) on how to do that. we just like to make sure nothing is breaking + all tests are passing. 

will ✅  once that's done!

**Comment by ITZSHOAIB** [comment]
> this looks great! could you open a PR with the `gamut-kit` alpha package? there are instructions [here](https://www.notion.so/codecademy/Bumping-gamut-kit-in-the-monorepo-b6107ee2b38f44cca176a967bdf67747#60ec17293542435cb56fb19f7679eea9) on how to do that. we just like to make sure nothing is breaking + all tests are passing.
> 
> will ✅ once that's done!

Hi @dreamwasp, here's a PR that uses this alpha version: https://github.com/codecademy-engineering/mono/pull/7110

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.2.0-alpha.7916db.0<br/>@codecademy/gamut-kit@0.6.435-alpha.7916db.0<br/>@codecademy/styleguide@67.1.5-alpha.7916db.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66bb4c78450da30a8965c5ad--gamut-preview.netlify.app>https://66bb4c78450da30a8965c5ad--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66bb4c78450da30a8965c5ad>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.2.0-alpha.372732.0<br/>@codecademy/gamut-kit@0.6.435-alpha.372732.0<br/>@codecademy/styleguide@67.1.6-alpha.372732.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66bcb41e665218ba5fb2f0c3--gamut-preview.netlify.app>https://66bcb41e665218ba5fb2f0c3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66bcb41e665218ba5fb2f0c3>Deploy Logs</a></p>

**Pull request merged**